axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/invoice/generator/InvoiceLineGeneratorSupplyChain.java

```diff
@@ -117,14 +117,17 @@ protected InvoiceLineGeneratorSupplyChain(
       this.taxLine = saleOrderLine.getTaxLine();
       this.discountTypeSelect = saleOrderLine.getDiscountTypeSelect();
       this.isTitleLine = saleOrderLine.getTypeSelect() == SaleOrderLineRepository.TYPE_PACK;
+      this.unit = saleOrderLine.getUnit();
+      this.qty = saleOrderLine.getQty();
```

Comment:
Qty should not get from saleOrderLine or purchaseOrderLine but depends of stockMoveLine. We can have partial stock move.